src/osd/ReplicatedBackend.cc

```diff
@@ -1971,6 +2072,9 @@ int ReplicatedBackend::build_push_op(const ObjectRecoveryInfo &recovery_info,
     new_progress.data_complete = true;
     if (stat)
       stat->num_objects_recovered++;
+  } else if (progress.first && progress.omap_complete) {
+    // If omap is not changed, we need recovery omap when recovery cannot be completed once
+    new_progress.omap_complete = false;
```

Comment:
Why do we need this? What limitation in the code make this logic necessary?

Reply:
If the object needs to be recovery multiple times, a temp object will be created, so even if the omap has not changed, we need to fill the omap of the temp object in submit_push_data.
For the case where multiple recovery is required and omap has not change , we can also populate the temp object with the omap of the need recovery object.